scripts/cat/cats.py

```diff
@@ -412,8 +412,14 @@ def die(self, body: bool = True):
 
         May return some additional text to add to the death event.
         """
-        self.injuries.clear()
-        self.illnesses.clear()
+        if self.status == 'leader' and 'pregnant' in self.injuries and game.clan.leader_lives > 0:
+            self.illnesses.clear()
+            for injury in self.injuries:
+                if injury != 'pregnant':
+                    del self.injuries[injury]
```

Comment:
Rather than using `del`, I suggest dictionary comprehension. This works, but it's very nested. 

replace lines 417, 418, and 419 with 
`self.injuries = { key : value for (key, value) in self.injuries.items() if key == 'pregnant'}

Reply:
oh thank you I had no idea what I was doing